bpf/bpf_wireguard.c

```diff
@@ -270,6 +270,9 @@ int cil_from_wireguard(struct __ctx_buff *ctx)
 
 	bpf_clear_meta(ctx);
 
+	/* mark packet as decrypted by wireguard */
+	ctx->mark = MARK_MAGIC_DECRYPT;
+
```

Comment:
```suggestion
#if defined(HAVE_ENCAP) && (!defined(ENABLE_NODEPORT) || !defined(ENABLE_NODE_ENCRYPTION))
		/* In native routing mode we want to deliver packets to local endpoints
 		 * straight from BPF, without passing through the stack.
 		 * This matches overlay mode (where bpf_overlay would handle the delivery)
 		 * and native routing mode without encryption (where bpf_host at the native
 		 * device would handle the delivery).
 		 *
 		 * When WG & encrypt-node are on, a NodePort BPF to-be forwarded request
 		 * to a remote node running a selected service endpoint must be encrypted.
 		 * To make the NodePort's rev-{S,D}NAT translations to happen for a reply
 		 * from the remote node, we need to attach bpf_host to the Cilium's WG
 		 * netdev (otherwise, the WG netdev after decrypting the reply will pass
 		 * it to the stack which drops the packet).
		 *
		 * Since neither of these conditions are met, we can return CTX_ACT_OK.
 		 */
		return CTX_ACT_OK;
#endif
```

Taken from the backport PR, this could be useful here, so that we keep the motivation around (previously in pkg/option/config.go, now removed in this commit).

Reply:
Maybe that'll fix the permanent CI failure I'm still facing. Something unexpected is still breaking with this change :thinking: